sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -1171,34 +1165,28 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
 
   test("Disable week-based date fields and quarter fields for parsing") {
 
-    def checkSparkUpgrade(c: Char): Unit = {
-      checkExceptionInExpression[SparkUpgradeException](
-        new ParseToTimestamp(Literal("1"), Literal(c.toString)).child, "3.0")
-      checkExceptionInExpression[SparkUpgradeException](
-        new ParseToDate(Literal("1"), Literal(c.toString)).child, "3.0")
-      checkExceptionInExpression[SparkUpgradeException](
-        ToUnixTimestamp(Literal("1"), Literal(c.toString)), "3.0")
-      checkExceptionInExpression[SparkUpgradeException](
-        UnixTimestamp(Literal("1"), Literal(c.toString)), "3.0")
+    def checkException[T <: Exception : ClassTag](c: String, onlyParsing: Boolean = false): Unit = {
```

Comment:
do we need the `onlyParsing` flag? we can get it by `Seq('E', 'F', 'q', 'Q').contains(c)`